Library/Homebrew/command.rb

```diff
@@ -0,0 +1,157 @@
+module Homebrew
+  class Command
+    attr_reader :command_name, :valid_options, :description, :help_output,
+      :man_output
+
+    def initialize_variables
+      @valid_options = []
+      @root_options = []
+      @argv = ARGV
```

Comment:
Let's do `ARGV.dup` here, just so we don't accidentally clobber the real `ARGV` if we end up modifying `@argv`.